python/ray/_raylet.pyx

```diff
@@ -361,10 +363,28 @@ cdef int prepare_actor_concurrency_groups(
         concurrency_groups.push_back(cg)
     return 1
 
-cdef prepare_args(
+cdef prepare_args_and_increment_put_refs(
         CoreWorker core_worker,
         Language language, args,
-        c_vector[unique_ptr[CTaskArg]] *args_vector, function_descriptor):
+        c_vector[unique_ptr[CTaskArg]] *args_vector, function_descriptor,
+        c_vector[CObjectID] *put_arg_ids):
+    try:
+        prepare_args_internal(core_worker, language, args, args_vector,
+                              function_descriptor, put_arg_ids)
+    except Exception as e:
+        # An error occurred during arg serialization. We must remove the
```

Comment:
Hmm, is it possible the refs haven't been added yet?

Reply:
We don't add to this list unless we've already incremented the local ref. Updated the variable name to clarify.